### DISCUSSION TITLE
Sync markdown from github

### DISCUSSION THREAD

**Discussion by benhowes** [outline/outline#2313] [Show and tell]
I've knocked up a quick script with github actions this morning so that I can sync markdown from github over to Outline. This is useful to me because I have some parts of my architecture documentation which is useful to have some version control (i.e. be able to have branches and pull requests), but I still want the documentation to be accessible to non-developer members of the team.

It's quite basic at the moment, but simply uses a Frontmatter block at the top of the source controlled markdown docs to set the metadata for the outline document (and store the doc id when created).

The only slightly unusual addition about my setup is that I really needed support for plantUML and mermaid diagrams to aid in explaining the architecture of our product. For this, I have a markdown parser step which creates kroki.io links for each of the detected diagram blocks.

See: https://github.com/benhowes/outline-sync-action

**Comment by tommoor** [comment]
Very cool, this is exactly the type of thing we hope to see built on the API 😄 – I'd love to one day include such functionality as a native integration

**Comment by januz** [comment]
@benhowes Nice! I just learned of outline and was searching for whether it was possible to sync a Github repo with markdown files. Your solution looks great. Are you still working on implementing the additional features you laid out in the README?

**Reply by benhowes** [reply]
I've not needed anything more just yet, so there's no additional stuff just yet
